From 94667901660eeb6ddf58bc13ae2e3ea0eb00a244 Mon Sep 17 00:00:00 2001
From: Tamal <tasaha@habana.ai>
Date: Fri, 15 Mar 2024 06:24:26 +0200
Subject: [PATCH] [SW-161727] gaudi3: mbist code cleanup

This patch reomoves unnecessary code related to
SPECIAL_MEM_ECC_ERR_MASK in function
gaudi3_mask_sram_ecc.

Architecture team confirmed, removed code
is not required for ecc error masking/unmasking
after sram mbist execution.

Change-Id: I2908e18629980cd46e2d39e51df295f8b0e61e94
Signed-off-by: Tamal <tasaha@habana.ai>
---

diff --git a/arch/gaudi3/arc/asic.c b/arch/gaudi3/arc/asic.c
index 5113f22..2009686 100644
--- a/arch/gaudi3/arc/asic.c
+++ b/arch/gaudi3/arc/asic.c
@@ -140,25 +140,13 @@
 
 static void gaudi3_mask_sram_ecc(bool enable)
 {
-	u32 val;
+	u32 val = 0;
 	int hdcore, cache;
 	u32 offset;
 
-	if (enable) {
-		regmap_writel(CPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK_SERR_M |
-				CPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK_DERR_M,
-				D0_CPU_IF_REG(mmCPU_IF_SPECIAL_MEM_ECC_ERR_MASK));
-		regmap_writel(CPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK_SERR_M |
-				CPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK_DERR_M,
-				D0_CPU_IF_REG(mmCPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK));
-
+	if (enable)
 		val = CACHE_SPECIAL_MEM_ECC_GLBL_ERR_MASK_SERR_M |
 				CACHE_SPECIAL_MEM_ECC_GLBL_ERR_MASK_DERR_M;
-	} else {
-		regmap_writel(0, D0_CPU_IF_REG(mmCPU_IF_SPECIAL_MEM_ECC_ERR_MASK));
-		regmap_writel(0, D0_CPU_IF_REG(mmCPU_IF_SPECIAL_MEM_ECC_GLBL_ERR_MASK));
-		val = 0;
-	}
 
 	/* Die 0 only at ppboot */
 	for (hdcore = 0; hdcore < GAUDI3_NUM_HDCORES_PER_DIE ; hdcore++) {
